src/Symfony/Component/HttpKernel/EventListener/ResponsePayloadListener.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Attribute\MapResponsePayload;
+use Symfony\Component\HttpKernel\Event\ViewEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Automatically serializes controller return values to JSON when decorated with #[MapResponsePayload].
+ */
+class ResponsePayloadListener implements EventSubscriberInterface
+{
+    public function __construct(
+        private ?SerializerInterface $serializer,
+    ) {
+    }
+
+    public function onKernelView(ViewEvent $event): void
+    {
+        $result = $event->getControllerResult();
+
+        // Skip if it's already a Response object
+        if ($result instanceof Response) {
```

Comment:
dead code. If the controller already returns a response, the HttpKernel does not dispatch the view event at all.

Reply:
Ah, I didn't know that. Thanks! I'll remove it.